spec/jobs/get_usps_proofing_results_job_spec.rb

```diff
@@ -478,6 +478,31 @@
           end
         end
 
+        context 'when the profile is deleted before proofing' do
+          before do
+            pending_enrollment.update!(profile_id: nil)
+            stub_request_passed_proofing_results
+          end
+
+          it 'updates the enrollment and ignores the non-existent profile' do
+            freeze_time do
+              pending_enrollment.update(
+                status_check_attempted_at: Time.zone.now - 1.day,
+                status_updated_at: Time.zone.now - 2.days,
+                )
+
+              expect { job.perform(Time.zone.now) }.not_to(raise_exception)
```

Comment:
I suppose instead of a safe navigation operator, you could wrap it in a `begin ... rescue` block? That seems too much, especially since you'd be doing it mostly to elicit a testable exception in a spec. So, maybe ignore me?

Reply:
An exception will fail the spec so all you have to do is leave the statement unwrapped and test the side effects after

```suggestion
              job.perform(Time.zone.now)
```

Reply:
@zachmargolis I thought about that a bit. The expectation is mostly pointless, but I kind of like it for its expressiveness.

Before I cleaned it up to look professional, the example was `it 'does not blow up'`. Realistically, that it no longer raises an exception is the main thing I wanted to test, even though, as you point out, the test would fail anyway if it did.

Does that make any sense, or is this still silly?

Reply:
I think it's just extra line noise, if the goal is to show that something can happen and continue cleanly, the test will still capture that regression.

Taking your approach to the extreme, if we had to assert *everywhere* that an exception didn't happen, our test suite would look very annoying

Reply:
OK, that's a good point. I initially felt like "It no longer blows up with an unexpected exception" was the changed behavior I wanted to test, but that is indeed just sort of a standard baseline expectation. 😄